test/test.js

```diff
@@ -387,31 +407,37 @@ describe('finalhandler(req, res)', function () {
 
     describe('when res.statusCode set', function () {
       it('should keep when >= 400', function (done) {
-        var server = http.createServer(function (req, res) {
+        var server = createServer(function (req, res) {
           var done = finalhandler(req, res)
           res.statusCode = 503
           done(new Error('oops'))
         })
 
-        request(server)
-          .get('/foo')
+        wrapper(request(server)
+          .get('/foo'))
           .expect(503, done)
       })
 
       it('should convert to 500 is not a number', function (done) {
-        var server = http.createServer(function (req, res) {
+        // http2 does not support non numeric status code
+        if (type === 'http2') {
+          done()
+          return
+        }
+
+        var server = createServer(function (req, res) {
           var done = finalhandler(req, res)
           res.statusCode = 'oh no'
           done(new Error('oops'))
         })
 
-        request(server)
-          .get('/foo')
+        wrapper(request(server)
+          .get('/foo'))
           .expect(500, done)
       })
 
       it('should override with err.status', function (done) {
-        var server = http.createServer(function (req, res) {
```

Comment:
Use `createServer` instead `http.createServer` to test using `http2` too, the same for tests below